.github/workflows/linux-build.yml

```
@@ -0,0 +1,267 @@
+        echo "CXX=g++" >> $GITHUB_ENV
+        
+    - name: Configure CMake
+      run: |
+        echo "Configuring build ${{ matrix.build_type }} with gcc"
+        cmake -B build/${{ matrix.build_type }} \
+              -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+              -DCCAP_BUILD_EXAMPLES=ON \
+              -DCCAP_BUILD_TESTS=ON
+    
+    - name: Build
+      run: cmake --build build/${{ matrix.build_type }} --config ${{ matrix.build_type }} --parallel $(nproc)
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2046:warning:1:91: Quote this to prevent word splitting

(shellcheck)